### ISSUE TITLE
Main menu button content and width problems

### ISSUE THREAD

**Issue by pierreporte** [OtterBrowser/otter-browser#1094]
I wanted to remove the label of the main menu button, so I went to the tab bar configuration window, selected the button and clicked on “Edit...”. The field to set a custom text is empty by default and there is no way to simply remove the label. This leads me to the following problematic behaviour.

I wanted to test the result of having a single space as the label. It was successful in the way that Otter accept this value, but the width of the button remains constant. Thinking that the button has simply a minimum width, I tested with the label “abcdefghijklmnopqrstuvwxyz” and the width remained the same. For this reason, I think that the width of the main menu button must adapt to the content of the button.

Screenshots:
![selection_511](https://cloud.githubusercontent.com/assets/2029853/14081838/f7266282-f50c-11e5-9b56-9fc502eb6693.png) (default)
![selection_513](https://cloud.githubusercontent.com/assets/2029853/14081842/fa9813ac-f50c-11e5-80c8-cc12e4afcb1c.png) (one-space label)
![selection_515](https://cloud.githubusercontent.com/assets/2029853/14081843/fd66bc78-f50c-11e5-902e-586f13a0a4d2.png) (alphabet label)

In the configuration window for the tab bar, when the main menu button has a custom label, the name in the list becomes the actual label text. It is currently problematic because when the custom text is visually empty (that is, it contains only space characters), then the line is invisible. If the button is moved to the end of the list (just after the closed tabs button), then the user will have a really hard time to find out that there is actually no bug. For this reason, The name of the button in the list should not be the custom label if there is one. If we want the custom label to be displayed, it should be done after (for example “Button menu (foo bar)” if the label is “foo bar”).

Screenshot with a one-space label:
![selection_514](https://cloud.githubusercontent.com/assets/2029853/14082030/0669c8aa-f50e-11e5-862f-33b65f06dbf8.png)

**To sum up:**

* Add a checkbox to disable the label of th main menu.
* Adapt the width of the button to its content.
* Do not fully replace the name of the button in the list with the label text.

**Comment by resbro** [comment]
I shortened up the "minimum size of the item" in the toolbar configuration until its too short to show the text. Worked for me. I had to play with it fer a while to get it shaped good.

**Comment by elebow** [comment]
I made a start at resizing the button automatically here: https://github.com/OtterBrowser/otter-browser/commit/26294e43de3ec5d84545bc788107697452ba15a3

![2016-03-29-22 59 19_1680x1050](https://cloud.githubusercontent.com/assets/7873686/14130629/72116c44-f602-11e5-9612-e14855475517.png)


It needs more work to correctly calculate the other buttons (Back, Forward, Reload, etc), and I'm still not sure how to collapse the space no longer taken by the button. I would think `ToolBarWidget::reload()` needs to be triggered somehow.

Moreover, it's not clear whether there should be a separate configurable minimum and maximum size. I think there should not be. Since the user can enter any text as the label, they can decide how long to make it, and the button should adjust to any width. Do others disagree?

**Comment by Emdek** [comment]
@pierreporte, constructor of menu button set standard text (translation of "Menu") and widget used that to calculate width, and only method used for painting used correct text in this case.
About disabling text completely... I guess that option to set button style per entry would be better, as it would both allow to easily disable label and greater customization.

@elebow, messing with size in that method without setting size hint is bad idea. ;-)
Existing sizing options should be enough, maximum size is handy in case of for example bookmarks bar with text beside icon and long titles.
